### ISSUE TITLE
First line of editor always read by screen reader

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#89231]
1. Turn on VoiceOver
2. Focus on an editor -> the first line of text is always read no matter where the cursor is

I believe this is a regression due to your change to use the first line for Untitled editors. Since this works just fine for VS Code stable.
However I see this for all editors, not only untitled ones.
@bpasero Can you take a look, and if you do not know what is going on feel free to assign back to me

fyi @ivanfetch

**Comment by bpasero** [comment]
@isidorn I cannot reproduce. I am adding this line `console.log(ariaLabel)` into https://github.com/microsoft/vscode/blob/365c034/src/vs/workbench/browser/parts/editor/textEditor.ts#L100 and it seems to not read the first line, only for untitled editors.

**Comment by isidorn** [comment]
Why don't you simply turn on Voice Over, just click Cmd + F5. 
Open Voice Over Utility > Speech > tick Mute Speech and Visuals > Panels and Menus > check Show caption panel

I can always reproduce it with Voice Over. Just swithc editor focus via Cmd + 1, Cmd + 2, or any other way.

**Comment by isidorn** [comment]
Actually I take it back. I broke this by changing the accessiblityPageSize to be automatic. So when we put 1000 lines in the contentArea that is too much for the VoiceOver.
Assignign back to me

**Comment by isidorn** [comment]
So the issue here is that VoiceOver always starts to read the document from the start whenever an editor gets focused. Before we were having only 10 lines in the hidden text area so VoiceOver would only read those 10 lines, and now since we have much mor (1000) VoiceOver would start from the start of the document.
We'll have to see how to outsmart VoiceOver with our hidden text area

fyi @alexdima

**Comment by alexdima** [comment]
@isidorn That is absurd... How does VoiceOver work when having a `<textarea>` in a website? It always reads the entire content? Even if it is very large? Surely reading the entire content of a textarea is not helpful...

**Comment by isidorn** [comment]
Yeah, agree this is absurd. I can reproduce this in Chrome with a `<textarea>` in Safari it works fine.
Thus I have created this issue in Chrome https://bugs.chromium.org/p/chromium/issues/detail?id=1045947

**Comment by MarcoZehe** [comment]
While testing if I could reproduce another issue, I found that in a C+ file at least, current Insider builds no longer read the first line for me at every instance, only when I am actually on that first line. So, something must have fixed this on the Chromium side in the meantime, or in Catalina‘s VoiceOver. Can anyone else still reproduce this? And if so, what steps do still reproduce this for you?

**Comment by isidorn** [comment]
@MarcoZehe I can still reproduce this issue.
Both in VSCode, Chrome, Safari etc. Might it be that there is some VoiceOver settings which controls this? Which you have set and I do not (I have default settings)

My os: Catalina 10.15.4
My steps: create two 10+ lines `.cpp` files, swithc between them using `Ctrl + Tab`, or `Ctrl + P`. Every time a file is focused VoiceOver reads content startign from the first line

@MarcoZehe let me know what you think and I can also file this bug to WebKit here https://bugs.webkit.org/enter_bug.cgi as @joanmarie recommends

**Comment by MarcoZehe** [comment]
Maybe I misunderstood the issue. It is normal that VoiceOver starts reading a newly focused text box or textarea. You can interrupt this and then arrow up and down normally. I thought this bug was about VoiceOver *always* reading the first line, even when on the second or third line and moving up and down with arrow keys. I am in a 40 k .mm file right now from the Mozilla source code, and I can read the file line by line without the first line being repeated. I have mostly default settings set, too.

**Comment by isidorn** [comment]
@marcozehe aha so you are arguing that this is VoiceOver expected behavior.
I just thought that it does not make sense to read the whole textArea which can contain 10000 characters, who would want to listen to that? Also other screen readers do not behave like this and thus I found it weird.
What is the benefit of hearing the whole text area instead of just the focused line?

**Comment by MarcoZehe** [comment]
Well, to be honest, it doesn't. It's behavior particular to VoiceOver which, in my opinion, indeed doesn't make sense for textareas or multiline text fields in general. But it is not an VS Code bug, that's what I was saying. And it doesn't always seem to do the whole thing, in my longer file, it seems to start reading the currently visible excerpt only. But again, I think it shouldn't do that, it should only read the current line. But that's a VoiceOver behavior change we could request, not a VS Code bug or something you can actively influence I believe.

**Comment by isidorn** [comment]
@marcozehe agree with you. Though we like to keep issues in our repository even if they should be fixed upstream.
Let's see if @cookiecrook responds since I believe he works on VoiceOver and I can also file an issue in the webkit bugs.

**Comment by isidorn** [comment]
This should be fixed with latest. If users are still facing this we can reopen the issue. Thanks
